tests/testthat/test-pipe_return_linter.R

```diff
@@ -0,0 +1,47 @@
+test_that("pipe_return_linter skips allowed usages", {
+  linter <- pipe_return_linter()
+
+  normal_pipe_lines <- trim_some("
+    x %>%
+      filter(str > 5) %>%
+      summarize(str = sum(str))
+  ")
+  expect_lint(normal_pipe_lines, NULL, linter)
+
+  normal_function_lines <- trim_some("
+    pipeline <- function(x) {
+      out <- x %>%
+        filter(str > 5) %>%
+        summarize(str = sum(str))
+      return(out)
+    }
+  ")
+  expect_lint(normal_function_lines, NULL, linter)
+
+  nested_return_lines <- trim_some("
+    pipeline <- function(x) {
+      x_squared <- x %>%
+        sapply(function(xi) {
+          return(xi ** 2)
+        })
+      return(x_squared)
+    }
+  ")
+  expect_lint(nested_return_lines, NULL, linter)
+})
+
+test_that("pipe_return_linter blocks simple disallowed usages", {
```

Comment:
I also checked this for lambdas and it works as expected. But I don't think we need to add tests for that.

Reply:
what do you mean by lambdas in this case? `|>`? see the comment in the R/ script -- native pipe is excluded because using return() there is a parser error

Reply:
No, I meant with purrr's mappers :)